File: GameServer/GameServerApp/GameServerApp/Proto/CreateLuaProto/BagListResponseProto.lua
--背包列表(返回)
BagListResponseProto = { ProtoCode = 50301 }

--这句是重定义元表的索引，就是说有了这句，这个才是一个类
BagListResponseProto.__index = BagListResponseProto;

function BagListResponseProto.New()
    local self = { }; --初始化self
    setmetatable(self, BagListResponseProto); --将self的元表设定为Class
    return self;
end


--发送协议
function BagListResponseProto.SendProto(proto)

    local ms = CS.LuaHelper.Instance:CreateMemoryStream();
    ms:WriteUShort(proto.ProtoCode);


    CS.LuaHelper.Instance:SendProto(ms:ToArray());
    ms:Dispose();
end


--解析协议
function BagListResponseProto.GetProto(buffer)

    local proto = BagListResponseProto.New(); --实例化一个协议对象
    local ms = CS.LuaHelper.Instance:CreateMemoryStream(buffer);


    ms:Dispose();
    return proto;
end

File: GameServer/GameServerApp/GameServerApp/Proto/CreateProto/BagListResponseProto.cs
//===================================================
//作    者：xxx
//创建时间：2019-05-18 15:13:23
//备    注：
//===================================================
using System.Collections;
using System.Collections.Generic;
using System;

/// <summary> 背包列表(返回) </summary>
public struct BagListResponseProto : IProto
{
    public ushort ProtoCode { get { return 50301; } }


    public byte[] ToArray()
    {
        using (MMO_MemoryStream ms = new MMO_MemoryStream())
        {
            ms.WriteUShort(ProtoCode);
            return ms.ToArray();
        }
    }

    public static BagListResponseProto GetProto(byte[] buffer)
    {
        BagListResponseProto proto = new BagListResponseProto();
        using (MMO_MemoryStream ms = new MMO_MemoryStream(buffer))
        {
        }
        return proto;
    }
}